pkg/sessionctx/stmtctx/stmtctx.go

```diff
@@ -1017,6 +957,8 @@ func (sc *StatementContext) ResetForRetry() {
 	sc.IndexNames = sc.IndexNames[:0]
 	sc.TaskID = AllocateTaskID()
 	sc.SyncExecDetails.Reset()
+	sc.WarnHandler.TruncateWarnings(0)
+	sc.ExtraWarnHandler.TruncateWarnings(0)
```

Comment:
I've confirmed with @time-and-fate that it's expected to also reset extra warning here. It's a mistake that it wasn't reset in `resetMuForRetry` before this PR.